src/main/java/org/luksza/gerrit/config/ConfigurationProvider.java

```diff
@@ -15,25 +15,50 @@
 package org.luksza.gerrit.config;
 
 import com.google.gerrit.extensions.annotations.PluginName;
+import com.google.gerrit.reviewdb.client.Project;
 import com.google.gerrit.server.config.PluginConfig;
 import com.google.gerrit.server.config.PluginConfigFactory;
+import com.google.gerrit.server.project.NoSuchProjectException;
 import com.google.inject.Inject;
 
 public class ConfigurationProvider {
-  private final PluginConfig pluginConfig;
+
+  private static final String JENKINSURL = "jenkinsUrl";
+  private static final String SELFNAME = "selfName";
+
+  private final PluginConfigFactory configFactory;
+  private final String pluginName;
+  private String defaultJenkinsUrl = "http://localhost:9090/";
+  private String defaultSelfName = "gerrit";
 
   @Inject
   ConfigurationProvider(
       PluginConfigFactory configFactory,
       @PluginName String pluginName) {
-    pluginConfig = configFactory.getFromGerritConfig(pluginName);
+    this.configFactory = configFactory;
+    this.pluginName = pluginName;
+
+    // Read default configuration from gerrit.config, for backward compatibility
+    PluginConfig pluginConfig = configFactory.getFromGerritConfig(pluginName);
+    defaultJenkinsUrl = pluginConfig.getString(JENKINSURL, defaultJenkinsUrl);
```

Comment:
either qualify this property with "this" as you did in lines 38 and 39 or remove "this" from above lines. please keep code consistent